natsjobs/item.go

```diff
@@ -28,6 +28,11 @@ type Item struct {
 	Options *Options `json:"options,omitempty"`
 }
 
+type JsonItem struct {
```

Comment:
Why?

Reply:
I found an another bug where header fields are lost during NATS message passing. The issue occurs because the Item struct has an unexported headers field that won't be included in JSON marshaling:
goCopytype Item struct {
    Job     string              `json:"job"`
    Ident   string             `json:"id"`
    Payload []byte             `json:"payload"`
    headers map[string][]string  // unexported, gets dropped during marshaling
    Options *Options           `json:"options,omitempty"`
}
Interestingly, the Context() method already handles headers correctly through a custom JSON struct, but this wasn't implemented for the main message passing to nats. To fix this, we can add:
goCopytype JsonItem struct {
    *Item
    Headers map[string][]string `json:"headers"`
}

This simple solution ensures headers are properly preserved when passing messages through NATS.

Reply:
This is not a simple solution, not a solution at all. Headers should be passed via NATS message headers and then populated via `unpack` method. I agree, that atm, headers are not properly passed, but they should not be passed like this.
The simple solution is the following: pass headers to the NATS message in the `Push` method. Get them in the listener `m.Headers()` and pass into the `Item`.

Reply:
Like this: https://github.com/roadrunner-server/nats/pull/182